packages/tradershub/src/features/cfd/screens/EnterPassword/EnterPassword.tsx

```diff
@@ -1,81 +1,61 @@
-    const title = PlatformDetails[platform].title;
+const EnterPassword = ({ onPasswordChange, password }: TEnterPasswordProps) => {
     const { isEU } = useRegulationFlags();
-
+    const { cfdState } = useCFDContext();
     const { data } = useActiveTradingAccount();
-    const accountType = data?.is_virtual ? 'Demo' : 'Real';
-    const marketTypeDetails = MarketTypeDetails(isEU);
+    const { closeModal } = useQueryParams();
 
+    const marketTypeDetails = MarketTypeDetails(isEU);
+    const { marketType = MarketType.ALL, platform = CFDPlatforms.MT5 } = cfdState;
+    const accountType = data?.is_virtual ? Category.DEMO : Category.REAL;
+    const title = PlatformDetails[platform].title;
     const marketTypeTitle =
         platform === PlatformDetails.dxtrade.platform ? accountType : marketTypeDetails[marketType]?.title;
 
     return (
-        <div className='flex lg:inline-flex lg:w-[400px] flex-col justify-center items-start rounded-default border-sm bg-system-light-primary-background'>
-            <Text weight='bold'>Enter your {title} password</Text>
-            <div className='flex flex-col gap-16 lg:gap-8 lg:pt-24'>
-                <Text size='sm'>
-                    Enter your {title} password to add a {title} {marketTypeTitle} account.
-                </Text>
-                <PasswordInput isFullWidth label={`${title} password`} onChange={onPasswordChange} value={password} />
-            </div>
-            {isDesktop && (
-                <div className='flex items-center justify-end w-full gap-8 pt-24'>
-                    <Button color='black' onClick={onSecondaryClick} size='lg' variant='outlined'>
-                        Forgot password?
-                    </Button>
-                    <Button
-                        disabled={!password || isLoading || !validPassword(password) || passwordError}
-                        isLoading={isLoading}
-                        onClick={onPrimaryClick}
-                        size='lg'
-                    >
-                        Add account
-                    </Button>
+        <React.Fragment>
+            <Modal.Header onRequestClose={closeModal}>
+                <Text weight='bold'>{`Enter your ${title} password`}</Text>
+            </Modal.Header>
+            <Modal.Body>
+                <div className='flex flex-col gap-16 lg:w-[400px] lg:gap-8 lg:pt-24 lg:px-24'>
+                    <Text size='sm'>
+                        Enter your {title} password to add a {title} {marketTypeTitle} account.
+                    </Text>
+                    <PasswordInput
+                        isFullWidth
+                        label={`${title} password`}
+                        onChange={onPasswordChange}
+                        value={password}
+                    />
                 </div>
-            )}
-        </div>
+            </Modal.Body>
+            <Modal.Footer>
+                <MT5PasswordFooter password={password} />
```

Comment:
This is just for MT5 ? how about others ?

Reply:
it does affect dxtrade since they are using the same component, have updated accordingly in the PR